channelserver/src/meta.rs

```diff
@@ -63,14 +63,14 @@ fn preferred_languages(alheader: String, default: &str) -> Vec<String> {
 // This rounds up from the dialect if possible.
 fn get_preferred_language_element(
     langs: &[String],
-    elements: BTreeMap<String, String>,
+    elements: BTreeMap<&str, &str>,
 ) -> Option<String> {
     for lang in langs {
         // It's a wildcard, so just return the first possible choice.
         if lang == "*" || lang == "-" {
-            return elements.values().next().map(std::borrow::ToOwned::to_owned);
+            return elements.values().next().map(|s| s.to_owned().to_owned());
```

Comment:
You can also deref first which is a little nicer IMO:

```suggestion
            return elements.values().next().map(|s| (*s).to_owned());
```